src/platform/plugins/shared/discover/public/application/main/discover_main_route.tsx

```diff
@@ -184,6 +304,11 @@ const DiscoverMainRouteContent = (props: SingleTabViewProps) => {
     for (const tabId of Object.keys(runtimeStateManager.tabs.byId)) {
       dispatch(internalStateActions.disconnectTab({ tabId }));
     }
+
+    // Clear conversation flyout active config when Discover unmounts
+    if (services.onechat) {
+      services.onechat.clearConversationFlyoutActiveConfig();
+    }
```

Comment:
Is this the only way that we can do this? Where we put the burden on consumers to clear what was previously stored in the flyout config state rather than us clearing it automatically when the active plugin ID changes / or however we detect this change in Kibana?

Reply:
i think so, seems to be a common pattern in kibana at least.